src/gazebo_vision_plugin.cpp

```diff
@@ -220,6 +220,118 @@ void VisionPlugin::rosOdomCallBack(const geometry_msgs::PoseWithCovarianceStampe
   }
 }
 
+void VisionPlugin::rosPoseCallBack(const geometry_msgs::PoseStampedConstPtr& odomMsg)
+{
+#if GAZEBO_MAJOR_VERSION >= 9
+  common::Time current_time = _world->SimTime();
+#else
+  common::Time current_time = _world->GetSimTime();
+#endif
+  double dt = (current_time - _last_pub_time).Double();
+
+  if (dt > 1.0 / _pub_rate) {
```

Comment:
As you can see, most of the parsing code is the same between callbacks. A function to parse the same code would be a more wise approach.